### DISCUSSION TITLE
Setup custom maven repository

### DISCUSSION THREAD

**Discussion by stavrogin** [renovatebot/renovate#8768] [Request Help]
Hi,
I would like to setup a custom maven repository, Nexus in my case, in addition to maven central, but I don't want to edit the project's pom.xml
I tried as explained [here](https://docs.renovatebot.com/java/) in a self-hosted Renovate so my config.js file that I inject in the docker container is like the following

```
module.exports = {
  repositories: [ "org/repo_name" ],
  dryRun : "false",
  endpoint: "https://github.xxxxx.com/api/v3",
  platform: "gitea",
  gitAuthor: "Flavio <flavio@flavio.com>",
  token: "***********",
  hostRules: [
    {
      hostType: "maven",
      baseUrl: "http://xxxxx.xxx.com:8081/nexus/content/groups/public/"
    }
  ]
}
```

Everything works fine but the Nexus dependencies in my pom are ignored. If I set it into my pom.xml instead, it works fine
Maybe is there something wrong in my config.js ?
Thanks

**Comment by rarkins** [comment]
You need to configure a `packageRules` entry with `registryURls` listing the registries you want to use. The `hostRules` entry you have should be removed as it does nothing.

**Reply by stavrogin** [reply]
Thank you @rarkins , it worked perfectly
